src/mlpack/core/data/dataset_info.hpp

```diff
@@ -37,7 +37,7 @@ class DatasetMapper
    */
   DatasetMapper(const size_t dimensionality = 0);
 
-  DatasetMapper(MapPolicy policy, const size_t dimensionality = 0);
+  DatasetMapper(PolicyType& policy, const size_t dimensionality = 0);
```

Comment:
Do you need the ability of implicit conversion?If not, maybe we could make the constructor as explicit?